service/src/main/java/bio/terra/cbas/initialization/BackfillOriginalWorkspaceIdBean.java

```diff
@@ -0,0 +1,58 @@
+package bio.terra.cbas.initialization;
+
+import bio.terra.cbas.config.CbasContextConfiguration;
+import bio.terra.cbas.dao.MethodDao;
+import bio.terra.cbas.dao.MethodVersionDao;
+import bio.terra.cbas.dao.RunSetDao;
+import bio.terra.cbas.util.BackfillOriginalWorkspaceIds;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BackfillOriginalWorkspaceIdBean {
+  private final Logger logger = LoggerFactory.getLogger(BackfillOriginalWorkspaceIdBean.class);
+  private final RunSetDao runSetDao;
+  private final MethodDao methodDao;
+  private final MethodVersionDao methodVersionDao;
+  private final CbasContextConfiguration cbasContextConfig;
+
+  public BackfillOriginalWorkspaceIdBean(
+      RunSetDao runSetDao,
+      MethodDao methodDao,
+      MethodVersionDao methodVersionDao,
+      CbasContextConfiguration cbasContextConfig) {
+    this.runSetDao = runSetDao;
+    this.methodDao = methodDao;
+    this.methodVersionDao = methodVersionDao;
+    this.cbasContextConfig = cbasContextConfig;
+  }
+
+  public void backfillOriginalWorkspaceIds() {
+    Boolean workspaceCreatedDateIsValid = false;
+
+    try {
+      // call this without assignment to make sure it's not empty,
+      // e.g. during local testing, or due to an upstream error
+      cbasContextConfig.getWorkspaceCreatedDate();
+      workspaceCreatedDateIsValid = true;
+    } catch (java.time.format.DateTimeParseException e) {
+      logger.error(
+          "Aborting backfillOriginalWorkspaceIds; There was an error parsing workspaceCreatedDate: {}",
+          e.getMessage());
```

Comment:
Are we likely to ever look for this message and do anything about it?

It seems like an outcome that causes CBAS to fail to start up (and force us to investigate) might be nicer in the long run than not updating the database and potentially losing data if we run straight into the clone recovery despite the backfilling not happening